graph/pkg/service/v0/service.go

```diff
@@ -110,6 +115,23 @@ func NewService(opts ...Option) Service {
 		svc.httpClient = options.HTTPClient
 	}
 
+	roleService := options.RoleService
+	if roleService == nil {
+		roleService = settingssvc.NewRoleService("com.owncloud.api.settings", grpc.DefaultClient)
+	}
+	roleManager := options.RoleManager
+	if roleManager == nil {
+		m := roles.NewManager(
+			roles.CacheSize(1024),
+			roles.CacheTTL(time.Hour*24*7),
```

Comment:
I don't think we should cache the roles for so long. I'm not sure if we should cache them at all on the 'client' side.
This is another thing we need to figure out regarding permissions... :confused:

Reply:
I copied that from the `ocs` service. But  yeah. A cache TTL of  one week seems a "little" exaggerated. :scream: 

@C0rby  Any suggestions for a more realistic ttl?

Reply:
I've reduced this timeout to 1hour by default now. I haven't touched any of the other places where caches are configured with these huge timeouts though.